components/gitpod-db/src/typeorm/project-db-impl.ts

```diff
@@ -70,6 +70,16 @@ export class ProjectDBImpl implements ProjectDB {
         await repo.save(project);
     }
 
+    public async setProjectSettings(projectId: string, settings: ProjectSettings): Promise<void> {
```

Comment:
For these kind of methods we usually have something like `updatePartial(shape: Partial<Project> & Pick<Project, "id">)` to avoid the repetition. Having the not-exists-check in here is a good addition!

Reply:
Sure, thanks for the hint! Went with a `DeepPartial` (so that properties like `settings` can also be Partial themselves).